src/pepr/operator/controllers/config/config.ts

```diff
@@ -3,109 +3,276 @@
+      configLog.debug("Updating KubeAPI network policies based on change to kubeApiCidr");
+      await initAPIServerCIDR();
+    }
   }
 
-  // Handle changes to the kubeNodeCidrs
-  if (decodedConfigData.KUBENODE_CIDRS !== UDSConfig.kubeNodeCidrs) {
-    UDSConfig.kubeNodeCidrs = decodedConfigData.KUBENODE_CIDRS;
-    // This re-runs the "init" function to update netpols if necessary
-    log.debug("Updating KubeNodes network policies based on change to kubeNodeCidrs");
-    await initAllNodesTarget();
+  if (!areKubeNodeCidrsEqual(networking?.kubeNodeCIDRs, UDSConfig.kubeNodeCIDRs)) {
+    UDSConfig.kubeNodeCIDRs = networking?.kubeNodeCIDRs || [];
+    if (updateClusterResources) {
+      // This re-runs the "init" function to update netpols if necessary
+      configLog.debug("Updating KubeNodes network policies based on change to kubeNodeCidrs");
+      await initAllNodesTarget();
+    }
   }
 
-  if (
-    decodedConfigData.UDS_DOMAIN !== UDSConfig.domain ||
-    decodedConfigData.UDS_ADMIN_DOMAIN !== UDSConfig.adminDomain
-  ) {
-    UDSConfig.domain = decodedConfigData.UDS_DOMAIN;
-    UDSConfig.adminDomain = decodedConfigData.UDS_ADMIN_DOMAIN;
-    if (!UDSConfig.domain || UDSConfig.domain === "###ZARF_VAR_DOMAIN###") {
+  if (expose.domain !== UDSConfig.domain || expose.adminDomain !== UDSConfig.adminDomain) {
+    if (expose.domain && expose.domain !== "###ZARF_VAR_DOMAIN###") {
+      UDSConfig.domain = expose.domain;
+    } else {
       UDSConfig.domain = "uds.dev";
     }
-    if (!UDSConfig.adminDomain || UDSConfig.adminDomain === "###ZARF_VAR_ADMIN_DOMAIN###") {
+    if (expose.adminDomain && expose.adminDomain !== "###ZARF_VAR_ADMIN_DOMAIN###") {
+      UDSConfig.adminDomain = expose.adminDomain;
+    } else {
       UDSConfig.adminDomain = `admin.${UDSConfig.domain}`;
     }
     // todo: Add logic to handle domain changes and update across virtualservices, authservice config, etc
   }
 
   // Update other config values (no need for special handling)
-  UDSConfig.allowAllNSExemptions = decodedConfigData.UDS_ALLOW_ALL_NS_EXEMPTIONS === "true";
+  UDSConfig.allowAllNSExemptions = policy.allowAllNsExemptions === true;
+
+  configLog.info(
+    `${firstLoad ? "Loaded" : "Updated"} UDS Config based on uds-operator-config ClusterConfig${firstLoad ? "" : " change"}`,
+  );
+}
+
+// Loads the UDS Config on startup
+export async function loadUDSConfig() {
+  // Run in Admission and Watcher pods
+  if (process.env.PEPR_WATCH_MODE || process.env.PEPR_MODE === "dev") {
+    let cfg: ClusterConfig = {};
+    let cfgSecret: kind.Secret = {};
 
-  log.info("Updated UDS Config based on uds-operator-config secret changes");
+    try {
+      cfg = await K8s(ClusterConfig).Get("uds-cluster-config");
+      cfgSecret = await K8s(kind.Secret).InNamespace("pepr-system").Get("uds-operator-config");
+    } catch (e) {
+      configLog.warn(e, "Error while fetching cluster config and operator config secret");
+    }
```

Comment:
Should this log be a bit more descriptive? seems like it would be useful to know more.

```suggestion
    } catch (error) {
        const err = error as Error;
        configLog.warn({
            error: err.message,
            stack: process.env.NODE_ENV === 'development' ? err.stack : undefined,
            operation: 'fetchConfig'
        }, "Failed to load cluster configuration");
        throw new Error(`Configuration load failed: ${err.message}`);
    }
```

Reply:
Open to changing a bit, although I think it's worth providing some context here:
- this func is only called on startup of pepr
- we don't want to immediately throw errors from this, since the `uds-operator-config` secret might not exist (and that's okay)
- we DO throw errors to crash the process if the clusterconfig doesn't exist
- this should only fail if (1) the clusterconfig/secret don't exist or (2) kubeapi issues
- the log includes the error (`e`) so the full error is present in the log message already
- in general that error will essentially be the http response from the k8s api so we'd have to parse a bit differently looking at `e.data?.message` for example

I'm going to separate this out/clean this up a bit to be more clear what is happening, but I think if we just log the full error object we're probably okay here?

Reply:
reviewed IRL looks good to me